lib/elastic_apm/transaction.rb

```diff
@@ -52,13 +53,19 @@ def initialize(
       @default_labels = config.default_labels
 
       @sampled = sampled
+      @sample_rate = sample_rate
 
       @context = context || Context.new # TODO: Lazy generate this?
       if @default_labels
         Util.reverse_merge!(@context.labels, @default_labels)
       end
 
-      @trace_context = trace_context || TraceContext.new(recorded: sampled)
+      unless (@trace_context = trace_context)
+        @trace_context = TraceContext.new(
+          traceparent: TraceContext::Traceparent.new(recorded: sampled),
+          tracestate: TraceContext::Tracestate.new(sample_rate: sample_rate)
```

Comment:
I think you're missing the a special case, setting sample_rate=0 when the transaction is not sampled

Reply:
Done